sequence_processing_pipeline/Pipeline.py

```diff
@@ -64,14 +113,14 @@ def directory_check(self, directory_path, create=False):
                 raise PipelineError("directory_path '%s' does not exist." %
                                     directory_path)
 
-    def find_bcl_directories(self):
+    def find_bcl_directories(self, root_path):
         """
         Walk root directory and locate all scan directory root folders
         :return: list of BCL directories within root folder.
         """
         new_dirs = []
 
-        for root, dirs, files in os.walk(self.run_dir):
+        for root, dirs, files in walk(root_path):
```

Comment:
Just confirming that this is OK, if it is, why use root_path vs. self.run_dir?

Reply:
It used to be that way when we could expect the input path to be a directory of run_directories. However it doesn't have much use when the input path is expected to be a single run_directory. I repurposed it for use in unittests. It was handy for getting a list of valid run_directories, but it needed to be able to accept different paths.